apiserver/test/e2e/service_server_e2e_test.go

```diff
@@ -207,6 +207,75 @@ func TestDeleteService(t *testing.T) {
 	}
 }
 
+func TestUpdateRayService(t *testing.T) {
+	tCtx, err := NewEnd2EndTestingContext(t)
+	require.NoError(t, err)
+
+	tCtx.CreateComputeTemplate(t)
+	t.Cleanup(func() {
+		tCtx.DeleteComputeTemplate(t)
+	})
+
+	testServiceRequest := createTestServiceV2(t, tCtx)
+	serviceName := testServiceRequest.Service.Name
+	namespace := testServiceRequest.Namespace
+
+	t.Cleanup(func() {
+		tCtx.DeleteRayService(t, serviceName)
+	})
+
+	// Verify initial replica value
+	require.Equal(t, int32(1), testServiceRequest.Service.ClusterSpec.WorkerGroupSpec[0].Replicas)
+
+	// Manually construct a new ClusterSpec with only the updated field changed
```

Comment:
Nit: also suggest to call out only replica changed, which is hard to tell from code